src/blockdata/transaction.rs

```diff
@@ -689,6 +710,23 @@ impl SigHashType {
          }
      }
 
+     /// Read a 4-byte uint32 as a standard sighash type, returning an error if the type
+     /// is non standard.
+     pub fn from_u32_standard(n: u32) -> Result<SigHashType, NonStandardSigHashType> {
+         // See above the rationale for this mask
+         let mask = 0x1f | 0x80;
```

Comment:
I think we still have it incorrect :). 

What is the use of this function/API?  
It should tell the user whether sighash flag they are using is standard or not. Standardness rules say that we should only accept [0x01, 0x02, 0x03, 0x81, 0x82, 0x83]. But as stated, this will allow other values too. So, I think correct way to have `from_u32_standard` have mask `ff`, but `from_u32` have mask `9f`.

Reply:
Right, corrected